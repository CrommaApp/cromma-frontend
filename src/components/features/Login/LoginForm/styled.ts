import styled from 'styled-components';

export const LoginFormContainer = styled.form`
	width: 100%;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	padding: 10px 60px;
	overflow-y: scroll;
	margin-top: 4%;

	& > ul {
		width: 100%;
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
	}

	& > button {
		width: 100%;
		border: 1px solid #dddddd;
		background-color: white;
		padding: 3% 4%;
		border-radius: 4px;
		color: #888888;
		margin-top: 12%;
	}
`;

export const LoginFormInput = styled.li`
	width: 100%;
	margin: 10px 0;

	& > input {
		width: 100%;
		border: 1px solid #dddddd;
		border-radius: 4px;
		padding: 10px;
	}

	& > p {
		font-size: 0.7rem;
		color: #aaaaaa;
		margin: 0;
		padding: 4px;
	}
`;
