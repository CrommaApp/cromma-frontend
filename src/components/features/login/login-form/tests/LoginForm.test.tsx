import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import LoginForm from '@components/features/login/login-form';

describe('LoginForm', () => {
	let closeLoginModal: () => void;

	beforeEach(() => {
		closeLoginModal = jest.fn();

		render(<LoginForm closeLoginModal={closeLoginModal} />);
	});

	it('Snapshot', () => {
		const component = renderer.create(<LoginForm closeLoginModal={closeLoginModal} />);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
