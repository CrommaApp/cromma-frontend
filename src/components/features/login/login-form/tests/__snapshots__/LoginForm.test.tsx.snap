// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Snapshot 1`] = `
<form
  className="sc-bdvvtL zRvut"
  onSubmit={[Function]}
>
  <ul>
    <li
      className="sc-gsDKAQ exJJSh"
    >
      <label
        className="a11y-hidden"
        htmlFor="login_input_id"
      >
        ID
      </label>
      <input
        aria-errormessage="id_error_msg"
        aria-invalid={false}
        aria-required="true"
        id="login_input_id"
        onChange={[Function]}
        placeholder="id"
        type="text"
        value=""
      />
    </li>
    <li
      className="sc-gsDKAQ exJJSh"
    >
      <label
        className="a11y-hidden"
        htmlFor="login_input_password"
      >
        Password
      </label>
      <input
        aria-errormessage="password_error_msg"
        aria-invalid={false}
        aria-required="true"
        id="login_input_password"
        onChange={[Function]}
        placeholder="passowrd"
        type="password"
        value=""
      />
    </li>
  </ul>
  <button
    type="submit"
  >
    로그인
  </button>
</form>
`;
